systers_portal/blog/views.py

```diff
@@ -389,3 +410,42 @@ def get_context_data(self, **kwargs):
                                                  slug=self.kwargs['slug'])
         context['tag_type'] = "Resource Type"
         return context
+
+
+class UserPinView(LoginRequiredMixin, ListView):
+    template_name = "blog/post.html"
+    model = Resource
+
+    def post(self, request, resource_slug, slug):
+        if request.method == "POST":
+            community = get_object_or_404(Community, slug=slug)
+            resource = get_object_or_404(Resource, community=community,
+                                         slug=resource_slug)
+            user = SystersUser.objects.get(user=self.request.user)
+            user_pins = UserPins.objects.filter(user=user)
+            if user_pins:
+                user_pins.first().add_pin(resource)
+            else:
+                user_pins = UserPins.objects.create(user=user)
+                user_pins.save()
```

Comment:
No need to save right?

Reply:
@SanketDG I don't think it's needed. Adding/removing is saved by default as far as I know!

Reply:
yes no need to save. Removed